programs/fileio.c

```diff
@@ -91,6 +91,32 @@ static clock_t g_time = 0;
 
 #define MIN(a,b)    ((a) < (b) ? (a) : (b))
 
+#if defined(_MSC_VER) && _MSC_VER >= 1400
+#   define LONG_SEEK _fseeki64
+#elif defined(__MINGW32__) && !defined(__STRICT_ANSI__) && !defined(__NO_MINGW_LFS) && defined(__MSVCRT__)
+#   define LONG_SEEK fseeko64
+#elif defined(_WIN32) && !defined(__DJGPP__)
```

Comment:
Yes, this is fallback solution created with this cases in mind. Now after looking into [MSDN docs](https://msdn.microsoft.com/en-us/library/s3f49ktz(v=vs.80).aspx) I see that VC++ 2003 will fail on other code anyway (due to lack of unsigned long long), but MinGW in standards mode is an important configuration.